src/discussions/learners/LearnerPostsView.test.jsx

```diff
@@ -62,31 +74,68 @@ describe('LearnerPostsView', () => {
   });
 
-  describe('Basic', () => {
-    test('Reported icon is visible to moderator for post with reported comment', async () => {
-      axiosMock.onGet(`${courseConfigApiUrl}${courseId}/`).reply(200, {
-        has_moderation_privileges: true,
-      });
-      axiosMock.onGet(`${courseConfigApiUrl}${courseId}/settings`).reply(200, {});
-      await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
-      await act(async () => {
-        renderComponent();
+  test('Reported icon is visible to moderator for post with reported comment', async () => {
+    await setUpPrivilages(axiosMock, store, true);
+    await waitFor(() => { renderComponent(); });
+
+    expect(container.querySelector('[data-testid="reported-post"]')).toBeInTheDocument();
+  });
+
+  test('Reported icon is not visible to learner for post with reported comment', async () => {
+    await renderComponent();
+    expect(screen.queryByTestId('reported-post')).not.toBeInTheDocument();
+  });
+
+  test('Learner title bar should display a title bar, a learner name, and a back button', async () => {
+    await renderComponent();
+
+    const titleBar = container.querySelector('.discussion-posts:first-child');
+    const learnerName = screen.queryByText('Activity for Abc123');
+    const backButton = screen.getByLabelText('Back');
+
+    expect(titleBar).toBeInTheDocument();
+    expect(learnerName).toBeInTheDocument();
+    expect(backButton).toBeInTheDocument();
+  });
+
+  test('Learner title bar should redirect to the learners list when clicking on the back button',
+    async () => {
+      await renderComponent();
+
+      const backButton = screen.getByLabelText('Back');
+
+      await act(() => fireEvent.click(backButton));
+      await waitFor(() => {
+        expect(lastLocation.pathname.endsWith('/learners')).toBeTruthy();
       });
-      expect(screen.queryAllByTestId('reported-post')[0]).toBeInTheDocument();
     });
 
-    test('Reported icon is not visible to learner for post with reported comment', async () => {
-      await renderComponent();
-      expect(screen.queryByTestId('reported-post')).not.toBeInTheDocument();
+  it('should display a post-filter bar and All posts sorted by recent activity text.', async () => {
+    await renderComponent();
+
+    const filterBar = container.querySelector('.filter-bar');
+    const recentActivity = screen.getByText('All posts sorted by recent activity');
+
+    expect(filterBar).toBeInTheDocument();
+    expect(recentActivity).toBeInTheDocument();
+  });
+
+  it(`should display a list of the interactive posts of a selected learner and the posts count
+     should be equal to the API response count.`, async () => {
+    await waitFor(() => {
+      renderComponent();
     });
+    const posts = await container.querySelectorAll('.discussion-post');
+
+    expect(posts).toHaveLength(2);
+    expect(posts).toHaveLength(Object.values(store.getState().threads.threadsById).length);
```

Comment:
are these both comparisons necessary?

Reply:
yes these are necessary, one for actually factory data length and another comparison is for the data in the redux